src/views/webviewController.ts

```diff
@@ -242,7 +254,30 @@ export default class WebviewController {
     }
   };
 
-  openWebview(context: vscode.ExtensionContext): Promise<boolean> {
+  openEditConnection = async ({
+    connection,
+    context,
+  }: {
+    connection: {
+      id: string;
+      name?: string;
+      connectionOptions: ConnectionOptions;
+    };
+    context: vscode.ExtensionContext;
+  }) => {
+    const webviewPanel = this.openWebview(context);
+
+    // Wait for the panel to open.
+    await new Promise((resolve) => setTimeout(resolve, 200));
```

Comment:
I wonder if we could avoid this. One quick thought is to listen to one of our own created event on webview to mark the webview as ready or maybe vscode api provides something?

Reply:
We definitely can, I like that idea. I'm going to merge in these changes and then open a follow up pr. As the diff in this pr is fairly large I want to be sure it's not going to cause large merge conflicts and block any other refactor work.